src/python/pants/backend/jvm/tasks/resolve_shared.py

```diff
@@ -13,18 +13,39 @@
 class ResolveBase(object):
   """Common methods for both Ivy and Coursier resolves."""
 
-  def add_directory_digests_for_jars(self, jars):
-    """Get DirectoryDigests for jars and return them zipped with the jars.
+  def add_directory_digests_for_jars(self, targets_and_jars):
+    """For each target, get DirectoryDigests for its jars and return them zipped with the jars.
 
-    :param jars: List of pants.java.jar.jar_dependency_utils.ResolveJar
-    :return: List of ResolveJars.
+    :param targets_and_jars: List of tuples of the form (Target, [pants.java.jar.jar_dependency_utils.ResolveJar])
+    :return: list[tuple[(Target, list[pants.java.jar.jar_dependency_utils.ResolveJar])]
     """
+
+    targets_and_jars=list(targets_and_jars)
+
+    if not targets_and_jars:
+      return targets_and_jars
+
+    jar_paths = []
+    for target, jars_to_snapshot in targets_and_jars:
+      [jar_paths.append(fast_relpath(jar.pants_path, get_buildroot())) for jar in jars_to_snapshot]
```

Comment:
It's a little odd to construct this list just to throw it away... Should this just be a for loop?

Reply:
That's a big oopsie on my part.

Reply:
I'll solve it inline but I cannot test it locally, I'm on the run.